### ISSUE TITLE
Unclear how loss is calculated

### ISSUE THREAD

**Issue by Xavear** [regel/loudml#158]
Dear all,

For my bachelor graduation thesis I hope to use the loss function which is calculated during training to validate my models. However, it is unclear to me how this loss function is calculated. I only saw 1 post shortly discussing the MSE (Mean Square Error) but can anyone confirm this is the one that is used? If not, can anyone tell how it is calculated? Maybe via the MAPE?

Thanks in advance!
Thom

**Comment by regel** [comment]
Hi Thom,

mean_square_error is only one component. See file `donut.py` line 123. The other component is KL loss calculated on the VAE latent space data. Our current implementation differs from the original DONUT that parametrize the variance, whereas ours is using log_var. This leads to a different loss value.

```
def add_loss(model, W):
```

This variant of VAE loss is modified (calculation of beta) to discard the effect of abnormal data points during loss computation. This interesting idea is discussed in DONUT paper.
